WeWatch/WeWatch/Service/Webservice.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  Webservice.swift
+//  WeWatch
+//
+//  Created by Anton on 30/11/2024.
+//
+
+import Foundation
+
+enum AuthenticationError: Error {
+    case invalidCredentials
+    case custom (errorMessage: String)
+}
+
+struct LoginRequestBody: Codable {
+    let apikey: String
+    let pin: String
+}
+
+struct LoginResponse: Codable {
+    let status: String
+    let data: LoginData?
+}
+
+struct LoginData: Codable {
+    let token: String?
+}
+
+class Webservice {
+    func login(apikey: String, pin: String, completion: @escaping (Result<String, AuthenticationError>) -> Void) {
```

Comment:
If you have multiple arguments in function, you can "break" them to make code more readable. Something like:

```
func login(
    apikey: String,
    pin: String,
    completion: @escaping (Result<String, AuthenticationError>
) -> Void) {
```